### PR TITLE
feat(sortByTimeToRead): [TML-122] Add read time and sort by time to read filter for prospects

### PR DESCRIPTION
**Pull request by Herraj**
## Goal

Add time to read property and sort by time to read filter for prospects.

## Todos

- [x] Add and fix tests
- [x] Refactor logic to fix unnecessary re-renders

Tickets:

- [TML-122](https://getpocket.atlassian.net/browse/TML-112)

[TML-122]: https://getpocket.atlassian.net/browse/TML-122?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx [hunk 1]
```diff
@@ -611,10 +625,14 @@ export const ProspectingPage: React.FC = (): JSX.Element => {
   const onSortByPublishedDate = () => {
     // from not sorted to sorted
     if (!sortByPublishedDate) {
+      // if sortByTimeToRead is set, toggle it off
+      sortByTimeToRead && setSortByTimeToRead(false);
```
Comment (vazish) [inline src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx:629]: Didn't know you could do this!

File: src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx [hunk 2]
```diff
@@ -624,6 +642,31 @@ export const ProspectingPage: React.FC = (): JSX.Element => {
...
+
+      return;
+    }
+
+    // if sortByPublishedDate is set, toggle it off before sorting by time to read below
+    sortByPublishedDate && setSortByPublishedDate(false);
+
+    // create a copy of the unsorted prospects
+    // we don't want to mutate the original unsorted prospect array
+    const prospectsToSort = [...unsortedProspects];
```
Comment (vazish) [inline src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx:659]: Nice!

**Review by vazish — APPROVED** [review]
Looks good! Glad you figured out the double-render issue.

File: src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx [hunk 3]
```diff
@@ -89,6 +89,13 @@ export const ProspectingPage: React.FC = (): JSX.Element => {
           scheduledSurfaceGuid: currentScheduledSurfaceGuid,
         },
       },
+      onCompleted: (data) => {
+        // after this query has finished fetching, if no data and no errors returned
+        // show empty state
+        if (!data && !error) {
```
Comment (vazish) [inline src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx:95]: No errors?

**Review by vazish — APPROVED** [review]
Looks good!

File: src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx (hunk 3 above)
Comment (Herraj) [reply src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx:95]: replied in slack!

**Pull request merged**